docs/root/intro/arch_overview/http/http_filters.rst

```diff
@@ -148,3 +148,79 @@ the filter config name.
 for if and how it is utilized for every filter.
+
+.. _arch_overview_http_filters_route_based_filter_chain:
+
+Route based filter chain
+------------------------
+
+There is support for having different filter chains for different routes. There are two different modes for this:
+
+* Disabling parts of filters in the filter chain for specific routes.
+* Setting parts of the filters in the filter chain to be disabled by default and then enabling them for specific
+  routes.
+
+By default, the filter chain is the same for all routes and all filters are enabled. However, they
+can be disabled for specific routes by using the :ref:`FilterConfig <envoy_v3_api_msg_config.route.v3.FilterConfig>`
+and setting the :ref:`disabled field <envoy_v3_api_field_config.route.v3.FilterConfig.disabled>` in the
+per filter config map in the route configuration. See the
+:ref:`Route specific config <arch_overview_http_filters_per_filter_config>` section for more details.
+
+For example, given following http filter config:
+
+.. code-block:: yaml
+
+  http_filters:
+  - name: buffer
+    typed_config: { ... }
+  - name: lua
+    typed_config: { ... }
+
+Both the ``buffer`` and ``lua`` filters are enabled by default. If we want to disable the ``buffer`` filter
+for a specific route, we can set the per filter config map in the route configuration:
+
+.. code-block:: yaml
+
+  typed_per_filter_config:
+    buffer:
+      "@type": type.googleapis.com/envoy.config.route.v3.FilterConfig
+      disabled: true
+
+In addition, we can set a filter to be disabled by default by setting the :ref:`disabled field
+<envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpFilter.disabled>`
+in the HttpFilter configuration and then enable it for specific routes if needed.
+
+For example, given following http filter config:
+
+.. code-block:: yaml
+
+  http_filters:
+  - name: buffer
+    typed_config: { ... }
+    disabled: true
+  - name: lua
+    typed_config: { ... }
+    disabled: true
+
+Both the ``buffer`` and ``lua`` filters are disabled by default. If we want to enable these two filters
+for a specific route, we can set per filter config map in the route configuration:
+
+.. code-block:: yaml
+
+  typed_per_filter_config:
+    lua:
+      "@type": type.googleapis.com/envoy.extensions.filters.http.lua.v3.LuaPerRoute
+      name: my_lua_script
+    buffer:
+      # This configuration self is meaningless but legal. No route per filter config is loaded from
+      # it but will be treated as enable flag for the filter 'buffer'.
+      # This is useful for the case that we want to enable the filter for a specific route but don't
+      # want to load any route specific configuration for it.
+      "@type": type.googleapis.com/envoy.config.route.v3.FilterConfig
+        config:
+          "@type": type.googleapis.com/google.protobuf.Empty
+        is_optional: true
```

Comment:
From our practice case, this is a real requirement, to enable a filter but don't want to provide any override route config.

I am open to the specific solution. Any better idea would be super appreciated.

It is okay for me to remove this case from the doc. I write it here just in case somebody may have similar scenarios as ours.

Reply:
I think there should be support for enabling a per-route, but not by setting `is_optional` to true.
IMHO a better option will be just overriding the per-route config, and not supplying anything else, but I think we should get some buy-in from the other @envoyproxy/api-shepherds before explicitly stating this.

Reply:
> IMHO a better option will be just overriding the per-route config, and not supplying anything else

Sorry, but I didn't get it. Could you give a more specific example about how to do it?

By the way, I just removed this case from the doc until we could agree on something.

Reply:
Apologies, I didn't clearly stated it. The example you have at the moment (after removing the `is_optional`) is, IMHO, a much better approach - enabling by overriding but not setting anything specific. But before giving a green-light, would like to hear if there are other objections/ideas.

Reply:
Get it. And rather than google.protobuf.Empty, maybe a specific message as the place holder. 🤔

Reply:
Should I create a new message as place holder if other api shepherds have no other concern? cc @adisuissa